app/src/main/java/com/example/mystore/ui/screens/LoginScreen.kt

```diff
@@ -31,30 +36,39 @@ import androidx.compose.ui.text.input.PasswordVisualTransformation
 import androidx.compose.ui.text.input.VisualTransformation
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
-import com.example.mystore.hasDigits
-import com.example.mystore.isMixedCase
-import com.example.mystore.isOverEightCharacters
+import com.example.mystore.util.hasDigits
+import com.example.mystore.util.isOverSevenCharacters
+import com.example.mystore.model.LoginModel
+import com.example.mystore.model.LoginResponse
+import com.example.mystore.networking.ApiClient
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.json.JSONObject
+import retrofit2.Response
+
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun LoginScreen(navigateToProductDetailsScreen: () -> Unit = {}) {
     var emailField by remember { mutableStateOf("") }
     var isEmailValid by remember { mutableStateOf(false) }
     var passwordField by remember { mutableStateOf("") }
-    var passwordVisible = rememberSaveable { mutableStateOf(false) }
-    val rightPassword = "Password01"
-    val rightEmail = "test@abv.bg"
+    val passwordVisible = rememberSaveable { mutableStateOf(false) }
     var isValidInput by remember { mutableStateOf(false) }
+    var isButtonEnabled by remember { mutableStateOf(true) }
     val mContext = LocalContext.current
+    val coroutineScope = rememberCoroutineScope()
+    val apiClient = ApiClient(mContext)
```

Comment:
The networking shouldn't reside in the view, it has to be part of the viewmodel. That way you can actually inject it, as you have that part done via the DI in the previous homework. Technically you should have a little bit of a different hierarchy than what you got here.

ViewModel <- injected repository <- injected API client <- injected interceptor <- injected context (for the token storage)

The idea is to have a type of cascading dependecy flow -- the API client is coming to any repository it is needed as it is a singleton (same goes for repositories, but the idea still stands), repositories are added to whichever viewmodel needs them, and viewmodels are provided to whichever view needs them. This holds true for the interceptor and the token storage class, too. 

N.B. The `Context` (we'll discuss this a bit more in future lectures as it is a very important part of the Android ecosystem, just use the application one for now) object should be handled with great care, [hence it is provided by Hilt by default](https://developer.android.com/training/dependency-injection/hilt-android#predefined-qualifiers). Check the example of its usage, but keep in mind it can be provided by default even in the `Modules` file. 

It's a bit hard to describe the concepts over text, so if you're still unsure what this meant, message me or anyone else, we can go over it again in a different manner.